### ISSUE TITLE
Subtle numeric difference from OpenCV

### ISSUE THREAD

**Issue by jerry73204** [strawlab/opencv-ros-camera#3]
Hey! I wrote a [test](https://github.com/jerry73204/opencv-ros-camera-test) that compares the projected pixels against actual OpenCV implementation.

By randomly sampling parameters and input points, it reaches about 97.5% success rate. It shows the Rust implementation covers most cases. I look into the failing parts and it seems due to numeric computation issues.

Do you consider to import or refer to the test, or document potential failing scenarios?

**Comment by astraw** [comment]
Yes, I would be interested to hear more and potentially add the tests.

**Comment by astraw** [comment]
Ah I didn’t initially see you linked the test. I will look into this.

**Comment by astraw** [comment]
I can reproduce your results on my system - thanks. I haven't started digging in any further. Have you started looking at the failure cases to check what might be going on?

**Comment by jerry73204** [comment]
Not in deep dive. I collect the erroneous cases. They are always large in coordinates, for example `9631530000000.0`. I guess it's rounding error at some place and multiplied to the one place.

**Comment by jerry73204** [comment]
I see the difference in rounding error might not be avoidable in implementation. Once we can confirm it's a numerical issue but not a logical issue, it's fine to close this issue.

**Comment by astraw** [comment]
I think we can build this and OpenCV to print the exact values to high precision at each step and then we can see at exactly which step things get different. Ideally there would be no divergence for all possible inputs.
